app/models/user.rb

```diff
@@ -38,7 +38,7 @@ class User < ActiveRecord::Base
   def role?(base_role)
     return false unless role
 
-    ROLES.index(base_role) <= ROLES.index(role.to_sym)
+    ROLES.index(base_role) == ROLES.index(role.to_sym)
```

Comment:
I like the explicit (error) version I think. Shouldn't happen too much and the log would be waaaay more helpful then.